src/components/EditDataModal/EditDataModal.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import { Input, Button } from 'antd';
+import Modal from 'react-modal';
+
+function EditModalDialog(props) {
+    const [assignee, setAssignee] = useState(props.patentDetails.patentAssignees);
+    const [patentName, setPatentName] = useState(props.patentDetails.patentName);
+    const [patentFiled, setPatentFiled] = useState(props.patentDetails.patentFiled);
+    const [patentDate, setPatentDate] = useState(props.patentDetails.patentDate);
+    const [claimed, setClaimed] = useState(props.patentDetails.claimedResidues);
+    const [patentLegalOpinion, setPatentLegalOpinion] = useState(props.patentDetails.patentLegalOpinion);
+
+    const customStyles = {
+        content: {
+            top: '50%',
+            left: '50%',
+            right: 'auto',
+            bottom: 'auto',
+            marginRight: '-50%',
+            transform: 'translate(-50%, -50%)',
+            width: '50%',
+        }
+    };
+
+    const submitPatentDetails = () => {
+        props.patentDetails.patentAssignees = assignee;
```

Comment:
props should be immutable and not be reassigned. If you need to edit the original patentDetails from PatentVisualizer you should pass the setPatentDetails function as a callback to get called here.

Reply:
fixed this